src/java/picard/vcf/MendelianViolationDetector.java

```diff
@@ -0,0 +1,338 @@
+        Haploid_Denovo("Offspring variant genotype is expected to be haploid and does not match an expected parental allele."),
+        Haploid_Other("Offspring reference genotype is expected to be haploid and does not match an expected parental allele."),
+        Other("Other unclassified violation of allele transmission.");
+
+        private final String description;
+
+        MendelianViolation(final String desc) {
+            this.description = desc;
+        }
+
+        public String getDescription() {
+            return description;
+        }
+    }
+
+    @Override
+    public void accumulate(final VariantContext ctx) {
+        logger.record(ctx.getContig(), ctx.getStart());
+
+        final String variantChrom = ctx.getContig();
+        final int variantPos = ctx.getStart();
+
+        // Skip anything a little too funky
+        if (ctx.isFiltered()) return;
+        if (!ctx.isVariant()) return;
+        if (!ctx.isSNP()) return;
+        if (!ctx.isBiallelic()) return;
+        if (SKIP_CHROMS.contains(variantChrom)) return;
+
+        for (final MendelianViolationMetrics trio : trios) {
+            final Genotype momGt = ctx.getGenotype(trio.MOTHER);
+            final Genotype dadGt = ctx.getGenotype(trio.FATHER);
+            final Genotype kidGt = ctx.getGenotype(trio.OFFSPRING);
+
+            // Test to make sure:
+            //   1) That the site is in fact variant in the trio
+            //   2) that the offspring doesn't have a really wacky het allele balance
+            if (!isVariant(momGt, dadGt, kidGt)) continue;
+            if (kidGt.isHet()) {
+                final int[] ad = kidGt.getAD();
+                if (ad == null) continue;
+                final double minAlleleFraction = Math.min(ad[0], ad[1]) / (double) (ad[0] + ad[1]);
+                if (minAlleleFraction < MIN_HET_FRACTION) continue;
+            }
+
+            ///////////////////////////////////////////////////////////////
+            // Determine whether the offspring should be haploid at this
+            // locus and which is the parental donor of the haploid genotype
+            ///////////////////////////////////////////////////////////////
+            boolean haploid = false;
+            Genotype haploidParentalGenotype = null;
+
+            if (FEMALE_CHROMS.contains(variantChrom)) {
+                if (trio.OFFSPRING_SEX == Sex.Female) {
+                    haploid = false;
+                } else if (isInPseudoAutosomalRegion(variantChrom, variantPos)) {
+                    haploid = false;
+                } else if (trio.OFFSPRING_SEX == Sex.Male) {
+                    haploid = true;
+                    haploidParentalGenotype = momGt;
+                } else {
+                    continue;
+                }
+            }
+
+            // the PAR on the male chromosome should be masked so that reads
```

Comment:
good point.

On Sun, Jun 26, 2016 at 12:48 PM, Tim Fennell <notifications@github.com>
wrote:

> In src/java/picard/vcf/MendelianViolationDetector.java
> <https://github.com/broadinstitute/picard/pull/536#discussion_r68505751>:
>
> > +            Genotype haploidParentalGenotype = null;
> > +
> > +            if (FEMALE_CHROMS.contains(variantChrom)) {
> > +                if (trio.OFFSPRING_SEX == Sex.Female) {
> > +                    haploid = false;
> > +                } else if (isInPseudoAutosomalRegion(variantChrom, variantPos)) {
> > +                    haploid = false;
> > +                } else if (trio.OFFSPRING_SEX == Sex.Male) {
> > +                    haploid = true;
> > +                    haploidParentalGenotype = momGt;
> > +                } else {
> > +                    continue;
> > +                }
> > +            }
> > +
> > +            // the PAR on the male chromosome should be masked so that reads
>
> I don't know what IGV is using for it's reference...
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/broadinstitute/picard/pull/536/files/908511b058da20df5ab1aeb43ca52cf444c59523#r68505751>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACnk0ig4QXh1JVL0NbIngFLanISew0Trks5qPq1RgaJpZM4IfA5X>
> .
>